content/cucumber/step-definitions.md

```diff
@@ -1,154 +1,170 @@
+{{% /text %}}
+```ruby
 Given "I have $n cucumbers in my belly" do |cukes|
   # Some Ruby code here
 end
 ```
 
+{{% text "ruby" %}}
 In this case, the String is compiled to a Regular Expression behind the scenes: `/^I have (.\*) cucumbers in my belly$/`.
+{{% /text %}}
 
+## Steps
 
-## What is a _Step_?
-
-Then there are Steps. A Step is analogous to a method or function *invocation*.
+A Step is analogous to a method call or function invocation.
 
-Here is an example:
+For example:
 
 ```
 Given I have 93 cucumbers in my belly
 ```
 
-In this Step, you're "calling" the above Step Definition with one argument: the string `"93"`. 
+In this Step, you're "calling" the above Step Definition with one argument: the value `93`.
 
-## _Where_ are Steps?
-
-Steps are declared in your `features/\*.feature` files. 
+Steps are declared in your {{% text "ruby" %}}`features/\*.feature`{{% /text %}}{{% text "java" %}}`*.feature`{{% /text %}}{{% text "javascript" %}}`*.feature`{{% /text %}} files.
 
 
 ## How Steps and Step Definitions work together
 
 1. Cucumber matches a Step against a Step Definition's `Regexp`
-2. Cucumber gathers any captures or variables
-3. Cucumber passes them to the Step Definition's `Proc` (or “function”) and executes it
-
+2. Cucumber gathers any capture groups or variables
+3. Cucumber passes them to the Step Definition's {{% text "ruby" %}}`Proc` (or “function”){{% /text %}}{{% text "javascript" %}}function{{% /text %}}{{% text "java" %}}method{{% /text %}} and executes it
 
 Recall that Step Definitions start with a [preposition](http://www.merriam-webster.com/dictionary/given) or an [adverb](http://www.merriam-webster.com/dictionary/when) (**`Given`**, **`When`**, **`Then`**, **`And`**, **`But`**). and can be expressed in any of Cucumber's supported [spoken languages](/gherkin/spoken-languages/).
 
-All Step Definitions are loaded (and defined) before Cucumber starts to execute the plain text.  
+All Step Definitions are loaded (and defined) before Cucumber starts to execute the plain text in the feature file.
 
-Once plain text execution begins, for each Step, Cucumber will look for a registered Step Definition with a matching `Regexp`. If it finds one, it will execute its Proc, passing all capture groups from the Regexp as arguments to the Proc.
+Once execution begins, for each Step, Cucumber will look for a registered Step Definition with a matching `Regexp`. If it finds one, it will execute it, passing all capture groups and variables from the Regexp as arguments to the method or function.
 
 The specific preposition/adverb used has **no** significance when Cucumber is registering or looking up Step Definitions.
 
 Also, check out Multiline [Step Arguments](/gherkin/gherkin-reference/#step-arguments) for more info on how to pass entire tables or bigger strings to your Step Definitions.
 
-
-## Successful Steps
```

Comment:
Might be worth putting all of these under a heading saying that they are the various states of steps, what do you think?